users/purge-users.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python
+from __future__ import print_function
+
+import sys
+import re
+import os
+from subprocess import check_output
+import optparse
+
+import yaml
+
+def list_users():
+    users = check_output("cut -d: -f1 /etc/passwd", shell=True).split('\n')
+    return users
+
+def list_devs():
+    users = list_users()
+    devs = [user for user in users if get_authorized_keys_filepath(user)]
+    return devs
+
+def get_authorized_keys_filepath(username):
+    return os.path.exists(os.path.join('/', 'home', username, '.ssh', 'authorized_keys'))
```

Comment:
The `pwd` module returns the user's home dir too, so we don't have to assume it's under `/home`.